extensions/amp-timeago/0.2/amp-timeago.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {PreactBaseElement} from '../../../src/preact/base-element';
+import {Timeago} from './timeago';
+import {dict} from '../../../src/utils/object';
+import {isExperimentOn} from '../../../src/experiments';
+import {isLayoutSizeDefined} from '../../../src/layout';
+import {userAssert} from '../../../src/log';
+
+/** @const {string} */
+const TAG = 'amp-timeago';
+
+class AmpTimeago extends PreactBaseElement {
+  /** @override */
+  init() {
+    return dict({'cutoffText': this.element.textContent});
+  }
+
+  /** @override */
+  isLayoutSupported(layout) {
+    userAssert(
+      isExperimentOn(this.win, 'amp-timeago-v2'),
+      'expected amp-timeago-v2 experiment to be enabled'
+    );
+    return isLayoutSizeDefined(layout);
+  }
+}
+
+/** @override */
+AmpTimeago.Component = Timeago;
+
+/** @override */
+AmpTimeago.passthrough = true;
+
+/** @override */
+AmpTimeago.props = {
+  'datetime': {attr: 'datetime'},
+  'locale': {attr: 'locale'},
+  'cutoff': {attr: 'cutoff', type: 'number'},
```

Comment:
Should you also be adding `cutoffText` here?

Reply:
`cutoffText` isn't an attribute on amp-timeago. It identifies the `textContent` passed in through `init()`. In v1 the component just read this directly.